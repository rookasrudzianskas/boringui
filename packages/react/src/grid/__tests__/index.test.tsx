import React from "react";
import {mount} from "enzyme";

import Grid from "../index";

describe("Grid", () => {
  it("should render correctly", () => {
    const wrapper = mount(
      <Grid.Container>
        <Grid>test</Grid>
        <Grid>test</Grid>
      </Grid.Container>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("all breakpoint values should be supported", () => {
    const wrapper = mount(
      <Grid.Container lg={4} md={3} sm={2} xl={5} xs={1}>
        <Grid lg={4} md={3} sm={2} xl={5} xs={1}>
          test
        </Grid>
      </Grid.Container>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("css value should be passed through", () => {
    const wrapper = mount(
      <Grid.Container
        alignContent="center"
        alignItems="center"
        direction="column"
        justify="center"
        wrap="wrap"
      >
        <Grid alignContent="center" alignItems="center" direction="column" justify="center">
          test
        </Grid>
      </Grid.Container>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("decimal spacing should be supported", () => {
    const wrapper = mount(
      <Grid.Container gap={0.11123}>
        <Grid>test</Grid>
      </Grid.Container>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("nested components should be supported", () => {
    const wrapper = mount(
      <Grid.Container>
        <Grid>test</Grid>
        <Grid.Container>
          <Grid>test</Grid>
          <Grid.Container>
            <Grid>test</Grid>
            <Grid.Container>
              <Grid>test</Grid>
            </Grid.Container>
            ,
          </Grid.Container>
          ,
        </Grid.Container>
        ,
      </Grid.Container>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("there should be no responsive class when not set", () => {
    let wrapper = mount(<Grid.Container sm={2} />);

    expect(wrapper.find(".nextui-grid-item").at(0).hasClass("sm")).toBeTruthy();
    expect(wrapper.find(".nextui-grid-item").at(0).hasClass("xs")).not.toBeTruthy();
    wrapper = mount(<Grid.Container />);
    expect(wrapper.find(".nextui-grid-item").at(0).hasClass("sm")).not.toBeTruthy();
  });

  it("should work correctly when use alone", () => {
    const wrapper = mount(<Grid />);

    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should work correctly when size exceeds", () => {
    const wrapper = mount(
      <Grid.Container>
        <Grid xs={13}>test</Grid>
        <Grid xs={-1}>test</Grid>
      </Grid.Container>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });
});
