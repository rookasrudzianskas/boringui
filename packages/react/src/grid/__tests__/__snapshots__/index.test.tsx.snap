// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `Grid all breakpoint values should be supported 1`
] = `"<div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-icgwmtB-css nextui-grid-item xs sm md lg xl nextui-grid-container\\"><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-iddATLy-css nextui-grid-item xs sm md lg xl\\">test</div></div>"`;

exports[
  `Grid css value should be passed through 1`
] = `"<div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-ibomLZK-css nextui-grid-item nextui-grid-container\\"><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-idTBjfx-css nextui-grid-item\\">test</div></div>"`;

exports[
  `Grid decimal spacing should be supported 1`
] = `"<div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-ikiOHGz-css nextui-grid-item nextui-grid-container\\"><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-igNCIse-css nextui-grid-item\\">test</div></div>"`;

exports[
  `Grid nested components should be supported 1`
] = `"<div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-ijDEIix-css nextui-grid-item nextui-grid-container\\"><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-igNCIse-css nextui-grid-item\\">test</div><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-ijDEIix-css nextui-grid-item nextui-grid-container\\"><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-igNCIse-css nextui-grid-item\\">test</div><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-ijDEIix-css nextui-grid-item nextui-grid-container\\"><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-igNCIse-css nextui-grid-item\\">test</div><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-ijDEIix-css nextui-grid-item nextui-grid-container\\"><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-igNCIse-css nextui-grid-item\\">test</div></div>,</div>,</div>,</div>"`;

exports[
  `Grid should render correctly 1`
] = `"<div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-ijDEIix-css nextui-grid-item nextui-grid-container\\"><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-igNCIse-css nextui-grid-item\\">test</div><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-igNCIse-css nextui-grid-item\\">test</div></div>"`;

exports[
  `Grid should work correctly when size exceeds 1`
] = `"<div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-ijDEIix-css nextui-grid-item nextui-grid-container\\"><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-icxokNG-css nextui-grid-item xs\\">test</div><div class=\\"nextui-c-kRHeuF nextui-c-kRHeuF-ieSqvQu-css nextui-grid-item xs\\">test</div></div>"`;
